betfund_solicitor/sendgrid.py

```diff
@@ -0,0 +1,58 @@
+"""SendGrid V3 API wrapper class."""
+
+from python_http_client.exceptions import BadRequestsError
+from sendgrid import SendGridAPIClient
+
+from betfund_solicitor import Message
+
+
+class SendGrid:
+    """Class for handling SendGrid email operations.
+
+    See: https://github.com/sendgrid/sendgrid-python
+
+    Attributes
+    ----------
+    client : `sendgrid.SendGridAPIClient`
+        Client to connect to SendGrid API.
+    message : `betfund_solicitor.Message`
+        Message object handling encapsulation for desired email.
+    """
+
+    def __init__(self, message: Message, sendgrid_api_key: str):
+        """SendGrid initialization.
+
+        Params
+        ------
+        message : `betfund_solicitor.Message`
+            Message object handling encapsulation for desired email.
+        sendgrid_api_key : str
+            Credentials to connect to SendGrid API.
+        """
+        self.client = SendGridAPIClient(api_key=sendgrid_api_key)
+        self.message = message
+
+    def __repr__(self):
+        return f"<SendGrid ({self.message})>"
+
+    def send(self):
+        """Sends email using `sendgrid.SendGridAPIClient.mail.send.post`.
+
+        Returns
+        -------
+        dict
```

Comment:
while the return is literal `dict` it returns variable `response`

Reply:
I would also include raise conditions in docstring

Reply:
Right, but I could theoretically rerun `{'hello': 'world'}`.. in which case, that wouldn't be response.  [Reference](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html).

Reply:
The function returns a variable `response` or could raise an exceptions, therefore the docstring should include the variable it returns along with the type.

Examples in reference return raw literal `bool` and a raw `str` that was never assigned to a variable

Reply:
Similar to how `Params` looks

Reply:
Agree I need to include a Raises in docstring. The Return is of type dictionary, regardless of what it is named.. I'll add the Raises piece.